moduleFromListExporter.nix

```diff
@@ -0,0 +1,47 @@
+{ flake-utils-plus }: let
+
+  modulesFromListExporter = paths:
+    /**
+    Synopsis: modulesFromListExporter _paths_
+
+    paths:    [ <path> <path> ]
+
+    Returns an attribute set of modules from a list of paths by converting
+    the path's basename into the attribute key.
+
+    Example:
+
+    paths:    [ ./path/to/moduleA.nix ./path/to/moduleBfolder ]
+
+    {
+      moduleA = import ./path/to/moduleA.nix;
+      moduleBfolder = import ./path/to/moduleBfolder;
```

Comment:
While working on the docs for moduleFromListExporter, I was wondering if we would take good advice here and overload those with some metadata concerning either or a combination of:
- location
- ownership (by this flake's current world-wide unique commit hash)

Prospectively, those enable us to distinguish in a global namespace which modules to auto-export by default without the user in need to tell us.

@Pacman99 How could such metadata enrichment potentially look like at this position?

I'd very much favor to record ownership by a specific commit hash, here, since it seems universally useful and directly testable by a downstream user's flake's `self.rev`.

Reply:
```nix
{
  modulesFromListExporter = paths: owningRev:
}
```
(if that makes sense)

---

maybe also
```nix
{
  overlaysFromChannelsExporter = channels: owningRev:
}
```

Reply:
You could use __functor for this. The overlay/module can go in __functor and metadata can be part of the attrset. I would say save this for future work however.